src/Reflex/Patch/DMapWithReset.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -Wall #-}
+
+-- | 'Patch'es on 'DMap' that consist only of insertions (or overwrites) and deletions.
+module Reflex.Patch.DMapWithReset where
+
+import Reflex.Patch.Class
+
+import Data.Dependent.Map (DMap, GCompare (..))
+import qualified Data.Dependent.Map as DMap
+import Data.Semigroup (Semigroup (..))
+import Data.Constraint.Extras
+
+-- | A set of changes to a 'DMap'.  Any element may be inserted/updated or deleted.
+-- Insertions are represented as @'ComposeMaybe' (Just value)@,
+-- while deletions are represented as @'ComposeMaybe' Nothing@.
+newtype PatchDMapWithReset k p = PatchDMapWithReset { unPatchDMapWithReset :: DMap k (By p) }
```

Comment:
I don't see the connection with ComposeMaybe? Leftover from vanilla PatchDMap?

Reply:
The name "WithReset" - what is "Reset" in this case? 

I think there's a need for this type (also for normal Map), I've seen a few instances of this type in different places...

Reply:
I also didn't know what `Reset` meant. Even if it has some precedence somewhere out there, the fact that none of us understand it means we should probably change it.